examples/dynamodb/v1beta1/resourcepolicy.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: dynamodb.aws.upbound.io/v1beta1
+kind: ResourcePolicy
+metadata:
+  annotations:
+    meta.upbound.io/example-id: dynamodb/v1beta1/resourcepolicy
+  labels:
+    testing.upbound.io/example-name: example
+  name: example-resourcepolicy
+spec:
+  forProvider:
+    policy: |
+      {
+        "Version": "2012-10-17",
+        "Statement": [
+          {
+            "Effect": "Allow",
+            "Principal": {
+              "AWS": "arn:aws:iam::123456789012:root"
+            },
+            "Action": [
+              "dynamodb:BatchGetItem",
+              "dynamodb:BatchWriteItem",
+              "dynamodb:DescribeTable",
+              "dynamodb:GetItem",
+              "dynamodb:PutItem",
+              "dynamodb:Scan",
+              "dynamodb:UpdateItem",
+              "dynamodb:Query",
+              "dynamodb:DeleteItem"
+            ],
+            "Resource": [
+              "arn:aws:dynamodb:us-east-2:123456789012:table/example"
```

Comment:
Thanks @mbbush I will fix that.

On another topic I was able to run the project locally using `make run` add provider config and create dynamdb table in my aws account but as for resource based policy attached to that table it didn't worked, got this error:
`read resource request failed: parsing resource ID: arn: invalid prefix` 

I think that my changes in `config/dynamodb/config.go` are ignored when running locally since in [config/provider.go](https://github.com/crossplane-contrib/provider-upjet-aws/blob/4b4cac415e215870c7f04328b51a4751c28e706c/config/provider.go#L35) the import of the artifact is remotely and not locally.
That can explain why I still seeing synced value of false and the arn parsing error above:
```
NAME                                             SYNCED   READY   EXTERNAL-NAME   AGE   LABELS
resourcepolicy.dynamodb.aws.upbound.io/example   False                            30m   testing.upbound.io/example-name=example
```
And on another note how do you debug the project ? or add some logs to it ? it would have been helpful to understand why the error message that I got occurred (maybe the arn is empty or using the external name of the dynamodb table not sure...)

If I can't relay on local testing to work can I just make sure it passed in Uptest?

Please advise and thanks! for answering.

Reply:
The error message I see is in the status condition, complaining about being unable to observe the resource.

Look at https://github.com/crossplane-contrib/provider-upjet-aws/actions/runs/9395450353, possibly also at the provider pod logs which should be attached to the run as an artifact, which might have more details. 

For running locally, I recommend `make e2e UPTEST_EXAMPLES_LIST=...` There are one or two other env vars you'll have to set in order to get your credentials into the provider config, and your AWS account id into the substitution. I think they're documented in the up test repo? Or maybe just reading the comments in the makefile in this project? This builds the appropriate provider subpackages for the manifest selected, starts a kind cluster, installs crossplane, side loads the built provider images onto the cluster, installs a provider config, and runs uptest.

Reply:
Thanks, In order to look at the provider pod logs I will need to either run it in uptest locally (since it runs under crossplane k8 deployment) or check the uptest action in github logs , is that correct?
From what I saw when running out of cluster (using `make run) the logs are printed only to the console but they didn't helped much.

BTW saw that you trirgger this action: https://github.com/crossplane-contrib/provider-upjet-aws/actions/runs/9397396899
I finished all the commits would like to run the e2e completely ? (saw only debug job ran)

Reply:
I'm on-the-go at the moment and can't easily run the tests myself, but one thing I'd try would be `kubectl get -o yaml --watch ResourcePolicy` before starting your test, so you can see every revision of the object. I wonder if there's some issue with the way the dynamodb table's ARN is getting resolved, leading to an invalid value, that's putting the provider into an error state that it's failing to recover from as it should. 

Something else to try would be to create a ResourcePolicy with the full ARN of the dynamodb table in `spec.forProvider.resourceArn` and/or in the `crossplane.io/external-name` annotation from the very start, and see if that changes any behavior.

None of these would be proper fixes, but they might help shed some light on what's happening so I can suggest what a proper fix would be.

Reply:
Regarding the env vars, I may well have misquoted the pluralization of `UPTEST_EXAMPLES_LIST` (or is it `UPTEST_EXAMPLE_LIST`?), so go by whatever's in the makefile, not my comment. There's also another env var, `UPTEST_DATASOURCE_PATH`, which you should set to the path to a yaml file containing `aws_account_id: '123456789012'` or whatever your account id is.

Reply:
I set the arn specifically that didn't help as well (with `resourceArn`), I will try to use watch on the ResourcePolicy maybe that will shed some light.
I've starting to think that maybe the terraform code itself return some other value than just a pure arn which make the provider fail, I'm testing with terraform command to see if there something strange, though `status.atProvier.arn` for the table look ok on the console logs of Uptest.
Thanks again